### PR TITLE
JAVA-24274 Upgrade Keycloak for oauth-jwt/jwt-auth-server

### PR DESCRIPTION
**Pull request by anuragkumawat**

### CODE REVIEW

File: oauth-jwt/jwt-auth-server/src/main/resources/baeldung-realm.json [hunk 1]
```diff
@@ -936,6 +936,14 @@
       "display.on.consent.screen" : "true"
     }
   }, {
```
Comment (anastasiosioannidis) [inline oauth-jwt/jwt-auth-server/src/main/resources/baeldung-realm.json:938]: Just for my information: Why is this needed?

File: oauth-jwt/jwt-auth-server/src/main/resources/META-INF/services/org.keycloak.config.ConfigProviderFactory
Comment (anastasiosioannidis) [inline oauth-jwt/jwt-auth-server/src/main/resources/META-INF/services/org.keycloak.config.ConfigProviderFactory:1]: Just for my information: Why is this needed?

File: oauth-jwt/jwt-auth-server/src/main/resources/baeldung-realm.json (hunk 1 above)
Comment (anuragkumawat) [reply oauth-jwt/jwt-auth-server/src/main/resources/baeldung-realm.json:938]: This is a client setting - If enabled, users have to consent to client access.

File: oauth-jwt/jwt-auth-server/src/main/resources/META-INF/services/org.keycloak.config.ConfigProviderFactory
Comment (anuragkumawat) [reply oauth-jwt/jwt-auth-server/src/main/resources/META-INF/services/org.keycloak.config.ConfigProviderFactory:1]: This is to provide path of our implementation of keycloak's JsonConfigProviderFactory class.
